### ISSUE TITLE
Mount failing on path with - (hyphen)

### ISSUE THREAD

**Issue by Belfine** [trapexit/mergerfs#640]
General description
===================
Not able to mount on path with - (hyphen) in path. In my example the path is "/mnt/b3d0989c-6015-4a48-b362-f8e4fc5c76a2/gdrive"

Expected behavior
=================
I escape all the special charatathers so it should mount without issue.
Service name:"mnt-b3d0989c\x2d6015\x2d4a48\x2db362\x2df8e4fc5c76a2-gdrive.mount"

Actual behavior
===============
The mount fails with this error:
`systemd[1]: mnt-b3d0989cx2d6015x2d4a48x2db362x2df8e4fc5c76a2-gdrive.mount: Where= setting doesn't match unit name. Refusing.`

**Precise** steps to reproduce the behavior
===========================================

Create service file: /etc/systemd/system/mnt-b3d0989c\x2d6015\x2d4a48\x2db362\x2df8e4fc5c76a2-gdrive.mount

Insert config in service file:
```
[Unit]
Description = /gdrive MergerFS mount
RequiresMountsFor=/data

[Mount]
What = /mnt/b3d0989c-6015-4a48-b362-f8e4fc5c76a2/local:/mnt/b3d0989c-6015-4a48-b362-f8e4fc5c76a2/rclone
Where = /mnt/b3d0989c-6015-4a48-b362-f8e4fc5c76a2/gdrive
Type = fuse.mergerfs
Options = async_read=false,use_ino,allow_other,auto_cache,func.getattr=newest,category.action=all,category.create=ff

[Install]
WantedBy=multi-user.target
```
Start service

I got the service name by running `systemd-escape -p --suffix=mount "/mnt/b3d0989c-6015-4a48-b362-f8e4fc5c76a2/gdrive"`

If I mount in "/mnt/test" it works

System information
==================
Please provide as much of the following information as possible:

* [ ] mergerfs version: `mergerfs version: 2.28.1
FUSE library version: 2.9.7-mergerfs_2.28.0
fusermount version: 2.9.8
using FUSE kernel interface version 7.29`
* [ ] mergerfs settings: `async_read=false,use_ino,allow_other,auto_cache,func.getattr=newest,category.action=all,category.create=ff`
* [ ] Linux version: `Ubuntu 18.04`
* [ ] Versions of any additional software being used
* [ ] List of drives, filesystems, & sizes:

```
Filesystem         Size  Used Avail Use% Mounted on
udev               3.8G     0  3.8G   0% /dev
tmpfs              778M  3.2M  775M   1% /run
/dev/sda2          147G   58G   83G  42% /
tmpfs              3.8G   20K  3.8G   1% /dev/shm
tmpfs              5.0M     0  5.0M   0% /run/lock
tmpfs              3.8G     0  3.8G   0% /sys/fs/cgroup
/dev/loop2          57M   57M     0 100% /snap/powershell/34
/dev/loop1          54M   54M     0 100% /snap/lxd/10526
/dev/loop3          57M   57M     0 100% /snap/powershell/19
/dev/loop6          54M   54M     0 100% /snap/core18/782
/dev/loop5          90M   90M     0 100% /snap/core/6818
/dev/loop10         90M   90M     0 100% /snap/core/6673
/dev/loop11         36M   36M     0 100% /snap/gtk-common-themes/1198
/dev/loop12         55M   55M     0 100% /snap/lxd/10756
/dev/loop13         57M   57M     0 100% /snap/powershell/21
/dev/loop14         54M   54M     0 100% /snap/core18/941
/dev/loop4          54M   54M     0 100% /snap/core18/970
/dev/loop8          89M   89M     0 100% /snap/core/6964
/dev/loop15        152M  152M     0 100% /snap/gnome-3-28-1804/51
/dev/loop16        3.8M  3.8M     0 100% /snap/gnome-system-monitor/83
/dev/loop9          54M   54M     0 100% /snap/lxd/10601
/dev/sdb           492G  185G  282G  40% /mnt/b3d0989c-6015-4a48-b362-f8e4fc5c76a2
/dev/loop17        3.8M  3.8M     0 100% /snap/gnome-system-monitor/87
/dev/loop18        152M  152M     0 100% /snap/gnome-3-28-1804/55
tmpfs              778M   20K  778M   1% /run/user/126
tmpfs              778M   48K  778M   1% /run/user/1000
gdrive:            1.0P   29T  1.0P   3% /mnt/b3d0989c-6015-4a48-b362-f8e4fc5c76a2/gdrive
local:rclone       492G  185G  282G  40% /mnt/test
```

**Comment by trapexit** [comment]
Why would you escape them? I doubt systemd cares about hyphens and the error message doesn't match what you would expect.

There is definitely no problem with mergerfs with hyphens. Many many users including myself have paths with hyphens.

**Comment by Belfine** [comment]
Hi, first of all thanks for the quick response and I just want to warn that I am not very experienced with Linux.

When Googling the error "Where= setting doesn't match unit name. Refusing." I get it kept getting me results like [this](https://unix.stackexchange.com/questions/283442/systemd-mount-fails-where-setting-doesnt-match-unit-name) where they escape them. [Another example ](https://serverfault.com/questions/774960/how-to-name-systemd-mount-unit-properly)

**Comment by trapexit** [comment]
That has to do with the filename matching the escaped Where. I'm not really a systemd user so I'm not familiar with all this. Have you used that tool as described?

**Comment by trapexit** [comment]
Or have you checked with whatever project you're taking these instructions from?

**Comment by Belfine** [comment]
As far as I can tell systemd requires the escaped name. I am using [animosity22's rclone guide](https://github.com/animosity22/homescripts), specifically this [systemd script](https://github.com/animosity22/homescripts/blob/master/rclone-systemd/gmedia.mount).

I just realized though that it is possible to mount it through fstab which I might try as I have used that in the past. Sorry, as I said I am still very new to the Linux world.
